flake.nix

```diff
@@ -1,17 +1,65 @@
 {
   description = "CLI for Hosted Nix binary caches";
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    hls.url = "github:haskell/haskell-language-server/master";
+  };
 
-  outputs = { self }: let
-    systems = [ "x86_64-linux" "i686-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
-    forAllSystems = f: builtins.listToAttrs (map (name: { inherit name; value = f name; }) systems);
-  in
-    {
-      packages = forAllSystems (
-        system: {
-          cachix = import ./. { inherit system; };
-        }
-      );
+  outputs = { self, nixpkgs, flake-utils, hls }:
+    with flake-utils;
+    lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        hPkgs =
+          pkgs.haskell.packages."ghc924"; # need to match Stackage LTS version
+        # from stack.yaml resolver
 
-      defaultPackage = forAllSystems (system: self.packages.${system}.cachix);
-    };
+        myDevTools =
+          with hPkgs; [
+            ghc # GHC compiler in the desired version (will be available on PATH)
+            ghcid # Continous terminal Haskell compile checker
+            ormolu # Haskell formatter
+            hlint # Haskell codestyle checker
+            hoogle # Lookup Haskell documentation
+            hls.packages."x86_64-linux"."haskell-language-server-924"
+            implicit-hie # auto generate LSP hie.yaml file from cabal
+            retrie # Haskell refactoring tool
+            hspec-discover
+            releaser
+            hkgr
+            niv
+            stack-wrapped
+          ];
+
+        # Wrap Stack to work with our Nix integration. We don't want to modify
+        # stack.yaml so non-Nix users don't notice anything.
+        # - no-nix: We don't want Stack's way of integrating Nix.
+        # --system-ghc    # Use the existing GHC on PATH (will come from this Nix file)
+        # --no-install-ghc  # Don't try to install GHC if no matching GHC found on PATH
+        stack-wrapped = pkgs.symlinkJoin {
+          name = "stack"; # will be available as the usual `stack` in terminal
+          paths = [ pkgs.stack ];
+          buildInputs = [ pkgs.makeWrapper ];
+          postBuild = ''
+            wrapProgram $out/bin/stack \
+              --add-flags "\
+                --system-ghc \
+                --no-install-ghc \
+              "
+          '';
+        };
+      in
+      {
+        packages.default = pkgs.cachix;
+        devShells.default = pkgs.mkShell {
+          buildInputs = myDevTools;
+
+          # Make external Nix c libraries like zlib known to GHC, like
+          # pkgs.haskell.lib.buildStackProject does
+          # https://github.com/NixOS/nixpkgs/blob/d64780ea0e22b5f61cd6012a456869c702a72f20/pkgs/development/haskell-modules/generic-stack-builder.nix#L38
+          LD_LIBRARY_PATH = pkgs.lib.makeLibraryPath myDevTools;
+          shellHook = ''stack build'';
```

Comment:
This is too much magic, entering a shell shouldn't take long.

Reply:
fixed 77f9d53